### PR TITLE
feat: add base for Netlify CMS feature for blog posts

### PR DESCRIPTION
**Pull request by zeropaper**
This PR is a spike / experiment aimed to allow non-technical writers to make updates / changes on the https://www.ory.sh website.

**Changes:**

- [x] added Netlify CMS
- [x] consistently using `.mdx` for blog post files
- [x] document development
- document usage - needs this PR to be merged

Benefits of using Netlify CMS:
- no additional setup besides creating editor identities (Here: https://app.netlify.com/sites/wonderful-sinoussi-8532a7/identity)
- keeps content in a single place
- content can be edited with the CMS (for non-technical people) and by editing markdown files (for more technical people)
- for now, has only 1 "content type" (blog post) but is very easily extendable to others 
- content types (AKA collections) can very easily be extended

image
image
image

## Related Issue or Design Document

Related to ory-corp/marketing#4

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Comment by zeropaper** [comment]
@Benehiko (you were suggested by GH as potential reviewers)
In order to document the publishing process when using the CMS feature (and... let's be honest, address a few details), I would need this PR to be merged (so that the CMS config get deployed on ory.sh).

I could merge, I believe it is risk free and I made a single commit (to be easily reverted... just in case), but since it touches a number of files and for transparency... I'd like your :+1:  :)

File: gatsby-browser.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 // You can delete this file if you're not using it
 import 'normalize.css'
-import 'phosphor-icons/src/css/phosphor.css'
+// import 'phosphor-icons/src/css/phosphor.css'
```
Comment (Benehiko) [inline gatsby-browser.js:8]: I see the icons package is commented out. This breaks the design of the website, is it intentional to not have icons?

Ory.sh has icons
image

Your PR has no icons
image

File: README.md [hunk 2]
```diff
@@ -6,11 +6,28 @@ This is the [www.ory.sh](https://www.ory.sh) website!
...
+#### Blog Articles
+
+When editing files of the Git repository, the blog post files
+must use the `.mdx` extension and placed in the [`src/markdown/blog`](src/markdown/blog) folder.
+
+**TODO**
+
+- frontmatter fields
+- using Netlify CMS
+- taxonomy URLs
```
Comment (Benehiko) [inline README.md:28]: will these `todo` comments be addressed in this PR? Sorry, I'm a bit out of context here, but isn't this PR addressing the Netlify CMS?

File: gatsby-node.js [hunk 3]
```diff
@@ -10,6 +13,78 @@ const trimLeft = (s, charlist) => {
...
+        component: path.resolve(`src/templates/tags.js`),
+        context: {
+          tag,
+        },
+      })
+    })
+  })
+}
+
+*/
```
Comment (Benehiko) [inline gatsby-node.js:87]: delete this if not used

**Review by Benehiko — CHANGES_REQUESTED** [review]
This looks quite good! This will really help with pushing content to the website.
I just have a few questions and suggestions:

- How does the CMS authenticate the admins? where do we create them?
- We probably need a guide in the readme to show people the exact admin portal URL and images on how to create a new blogpost etc.
- The icons are missing and some of the code that is commented out needs to be cleaned up.

File: gatsby-node.js (hunk 3 above)
Comment (zeropaper) [reply gatsby-node.js:87]: Damn... sorry about that...

File: README.md (hunk 2 above)
Comment (zeropaper) [reply README.md:28]: I need the PR to be first merged to actually add relevant / precise documentation.
But it will be done.

**Comment by zeropaper** [comment]
@Benehiko Thanks a bunch for your review!

> I just have a few questions and suggestions:
>
> How does the CMS authenticate the admins? where do we create them?
> We probably need a guide in the readme to show people the exact admin portal URL and images on how to create a new > blogpost etc.
> The icons are missing and some of the code that is commented out needs to be cleaned up.

- The authentication, for now, will be managed on Netlify Identity (sad... because we should do it, obviously :) )
- In order to write the guide with the relevant information, I would need to merge that (since the local development / usage) of the CMS is slightly different, but sure enough, there will be notes and guides in the README
- I encountered some issues with old deps, commented that line and forgot about it.... but I fixed it. Thanks!

**Review by Benehiko — APPROVED** [review]
LGTM! :)

**Pull request merged**